src/Core/src/Platform/Android/ComponentCallbacks.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Android.App;
+using Android.Content;
+using Android.Content.Res;
+using Microsoft.Maui.LifecycleEvents;
+
+namespace Microsoft.Maui
+{
+	class ApplicationComponentCallbacks : ComponentCallbacks
+	{
+		readonly Application _application;
+
+		public ApplicationComponentCallbacks(Application application) =>
+			this._application = application;
+
+		public override void OnConfigurationChanged(Configuration newConfig) =>
+			MauiApplication.Current.Services?.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationConfigurationChanged>(del => del(_application, newConfig));
+
+		public override void OnLowMemory() =>
+			MauiApplication.Current.Services?.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationLowMemory>(del => del(_application));
+
+		public override void OnTrimMemory(TrimMemory level) =>
+			MauiApplication.Current.Services?.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationTrimMemory>(del => del(_application, level));
+	}
+
+	class ActivityComponentCallbacks : ComponentCallbacks
+	{
+		readonly WeakReference<Activity> _activity;
+
+		public ActivityComponentCallbacks(Activity activity) =>
+			this._activity = new WeakReference<Activity>(activity);
+
+		public override void OnConfigurationChanged(Configuration newConfig)
+		{
+			if (!_activity.TryGetTarget(out var target))
+				return;
+			MauiApplication.Current?.Application.ThemeChanged();
```

Comment:
Does this belong here? Seems like this should maybe go in the host builder extension: https://github.com/dotnet/maui/blob/main/src/Core/src/Hosting/LifecycleEvents/AppHostBuilderExtensions.Android.cs#L19

Or should this be called here at all?  What's the order of invocation and difference between the activity and application configuration changed being called?  Will they both be called?